tests/unity/unity_cfg.yaml

```diff
@@ -30,6 +31,22 @@
         'uint32_t': 'HEX32'
         'int64_t': 'INT64'
         'uint64_t': 'HEX64'
+    # Disable automatic mock generation for some functions:
+    # nrf_modem_at_printf:
+    #   CMock does not have support for testing parameters of variadic functions. Hence we have
+    #   to make CMock not generate mock for nrf_modem_at_printf and write our own mock/stub for
+    #   this function.
```

Comment:
>I'm planning to use the mock of nrf_modem_at_printf in SMS tests in PR #6314 and I guess this will get broken?

I see that you are only passing one parameter to this function in your implementation (either AT_SMS_SUBSCRIBER_REGISTER or AT_SMS_SUBSCRIBER_UNREGISTER). So you can use CMock generated mock for testing this. Only if you pass more than one parameter, then you will need to write your own mock.

> the real solution would be to implement nrf_modem_at_mock library to support full mocking of nrf_modem_at functions instead of all different tests doing their own stuff...

+1 . An idea would be to have a common mock that calls a stub with `fmt` and `va_list  args` . The stub needs to be implemented by the test because each test may have different expectations for the parameters.

Reply:
> I see that you are only passing one parameter to this function in your implementation (either AT_SMS_SUBSCRIBER_REGISTER or AT_SMS_SUBSCRIBER_UNREGISTER). So you can use CMock generated mock for testing this. Only if you pass more than one parameter, then you will need to write your own mock.

But is the mock available for me if this PR adds `nrf_modem_at_printf()` to some ignore list in `tests/unity/unity_cfg.yaml`?

> +1 . An idea would be to have a common mock that calls a stub with `fmt` and `va_list args` . The stub needs to be implemented by the test because each test may have different expectations for the parameters.

I guess we could implement a generic solution which would work for many use cases. CMock has done it for other functions already so we could do it for functions with variadic arguments. Haven't thought about it completely so not sure if it would work but I'm afraid many libraries would need to implement similar mock unless we have a generic one.

Reply:
>But is the mock available for me if this PR adds nrf_modem_at_printf() to some ignore list in tests/unity/unity_cfg.yaml?

Ah no. Mock generation is disabled to all unit test suites in nrf repo and I did not think about it when answered you here.  I am not sure if there is a way to disable mock generation for individual test suites. I will look into it.

Reply:
>CMock has done it for other functions already so we could do it for functions with variadic arguments.

Not sure what you mean here. Can you elaborate?

Reply:
@trantanen  I have now pushed a commit that solves the problem for both of us. I have only disabled mock generation for `nrf_modem_at_printf()` for my test suite. 

Related commit : https://github.com/nrfconnect/sdk-nrf/pull/6471/commits/a5c4997a4ac9e3410caf39757686ebddc12978c7

Reply:
This is fine now.

> > CMock has done it for other functions already so we could do it for functions with variadic arguments.
> 
> Not sure what you mean here. Can you elaborate?

We could implement some kind of generic mock library for nrf_modem_at which would be properly parameterized to handle various use cases we have. But your current solution is good enough for now.